qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -262,41 +293,38 @@ def clear(self):
         if self.search_displayed:
             self.cleared.emit()
         self.search_displayed = False
-        if self._nowrap_available:
-            self._reset_match_data()
+        if self._wrap_handler is not None:
+            self._wrap_handler.reset_match_data()
         self._widget.findText('')
 
-    def _message_wrap_prevented(self, top=True):
-        if top:
-            message.info("Search hit TOP")
-        else:
-            message.info("Search hit BOTTOM")
-
     def prev_result(self, *, result_cb=None):
         # The int() here makes sure we get a copy of the flags.
         flags = QWebEnginePage.FindFlags(int(self._flags))  # type: ignore
-        nowrap = self._nowrap_available and not self._flag_wrap
+        nowrap = (self._wrap_handler is not None and
+                  not self._wrap_handler.flag_wrap)
         if flags & QWebEnginePage.FindBackward:
-            if nowrap and self._active_match == self._total_matches:
-                self._message_wrap_prevented(top=False)
+            if (nowrap and self._wrap_handler.active_match ==
+                    self._wrap_handler.total_matches):
+                self._wrap_handler.message_wrap_prevented(top=False)
                 return
             flags &= ~QWebEnginePage.FindBackward
         else:
-            if nowrap and self._active_match == 1:
-                self._message_wrap_prevented(top=True)
+            if nowrap and self._wrap_handler.active_match == 1:
+                self._wrap_handler.message_wrap_prevented(top=True)
                 return
             flags |= QWebEnginePage.FindBackward
         self._find(self.text, flags, result_cb, 'prev_result')
 
     def next_result(self, *, result_cb=None):
-        if self._nowrap_available and not self._flag_wrap:
+        if self._wrap_handler is not None and not self._wrap_handler.flag_wrap:
             going_up = self._flags & QWebEnginePage.FindBackward
 
-            if going_up and self._active_match == 1:
-                self._message_wrap_prevented(top=True)
+            if going_up and self._wrap_handler.active_match == 1:
+                self._wrap_handler.message_wrap_prevented(top=True)
                 return
-            elif not going_up and self._active_match == self._total_matches:
-                self._message_wrap_prevented(top=False)
+            elif (not going_up and self._wrap_handler.active_match ==
+                  self._wrap_handler.total_matches):
+                self._wrap_handler.message_wrap_prevented(top=False)
```

Comment:
Haven't thought about it in detail, but maybe it'd make sense to move some more logic into the wrap handler from here, and then make `active_match` and `total_matches` private in it? Then you could do things like `self._wrap_handler.check_wrap(top=going_up)` and the wrap handler would do the rest.